jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -148,20 +141,87 @@ private static boolean isProgressFooterEnabled(Project project) {
 
     eventHandlers = makeEventHandlers(project, logger, singleThreadedExecutor);
   }
 
   @Override
-  public JavaLayerConfigurations getJavaLayerConfigurations() {
-    return javaLayerConfigurations;
+  public JibContainerBuilder getContainerBuilderWithLayers(RegistryImage baseImage) {
+    try {
+      if (isWarProject()) {
+        logger.info("WAR project identified, creating WAR image: " + project.getDisplayName());
+        Path explodedWarPath = GradleProjectProperties.getExplodedWarDirectory(project);
+        return JavaContainerBuilderHelper.fromExplodedWar(baseImage, explodedWarPath, appRoot);
+      }
+
+      JavaPluginConvention javaPluginConvention =
+          project.getConvention().getPlugin(JavaPluginConvention.class);
+      SourceSet mainSourceSet =
+          javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
+
+      FileCollection classesOutputDirectories =
+          mainSourceSet.getOutput().getClassesDirs().filter(File::exists);
+      Path resourcesOutputDirectory = mainSourceSet.getOutput().getResourcesDir().toPath();
+      FileCollection allFiles = mainSourceSet.getRuntimeClasspath();
+      FileCollection dependencyFiles =
+          allFiles
+              .minus(classesOutputDirectories)
+              .filter(file -> !file.toPath().equals(resourcesOutputDirectory));
+
+      JavaContainerBuilder javaContainerBuilder =
+          JavaContainerBuilder.from(baseImage).setAppRoot(appRoot);
+
+      // Adds resource files
+      if (Files.exists(resourcesOutputDirectory)) {
+        javaContainerBuilder.addResources(resourcesOutputDirectory);
+      }
+
+      // Adds class files
+      for (File classesOutputDirectory : classesOutputDirectories.filter(File::exists)) {
+        if (classesOutputDirectory.exists()) {
+          javaContainerBuilder.addClasses(classesOutputDirectory.toPath());
+        }
+      }
+      if (classesOutputDirectories.filter(File::exists).isEmpty()) {
+        logger.warn("No classes files were found - did you compile your project?");
+      }
+
+      // Adds dependency files
+      javaContainerBuilder.addDependencies(
+          dependencyFiles
+              .getFiles()
+              .stream()
+              .filter(File::exists)
+              .map(File::toPath)
+              .collect(Collectors.toList()));
+
+      return javaContainerBuilder.toContainerBuilder();
+
+    } catch (IOException ex) {
+      throw new GradleException("Obtaining project build output files failed", ex);
+    }
+  }
+
+  @Override
+  public ImmutableList<Path> getClassFiles() throws IOException {
+    // TODO: Consolidate with getContainerBuilderWithLayers
+    JavaPluginConvention javaPluginConvention =
+        project.getConvention().getPlugin(JavaPluginConvention.class);
+    SourceSet mainSourceSet = javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
+    FileCollection classesOutputDirectories = mainSourceSet.getOutput().getClassesDirs();
```

Comment:
From the code in `getContainerBuilderWithLayers()`, seems like this may return directories that don't exist. Will this (`DirectoryWalker(dir.toPath()).walk()`, I guess) work without any problem?

Reply:
I'll add a test.

Reply:
Actually yeah, this'll fail, I'll change it.